src/pudl/output/eia860.py

```diff
@@ -464,3 +485,616 @@ def ownership_eia860(pudl_engine, start_date=None, end_date=None):
+    generators and units still available in the dataframe!
+
+    Args:
+        gens_df (pandas.DataFrame): A generators_eia860 dataframe containing at
+            least the columns plant_id_eia and unit_id_pudl.
+
+    Returns:
+        pandas.DataFrame: Having two columns: plant_id_eia and max_unit_id_pudl
+        in which each row should be unique.
+
+    """
+    return (
+        gens_df[["plant_id_eia", "unit_id_pudl"]]
+        .drop_duplicates()
+        .groupby("plant_id_eia").agg({"unit_id_pudl": max})
+        .fillna(0)
+        .rename(columns={"unit_id_pudl": "max_unit_id_pudl"})
+        .reset_index()
+    )
+
+
+def assign_simple_unit_ids(
+    gens_df,
+    prime_mover_codes,
+    fuel_type_code_pudl=None,
+    label_prefix="single"
+):
+    """
+    Assign unique PUDL Unit IDs to all generators of a given prime mover type.
+
+    Calculate the maximum pre-existing PUDL Unit ID within each plant, and
+    assign each as of yet unidentified distinct generator within each plant
+    with an incrementing integer unit_id_pudl, beginning with 1 + the previous
+    maximum unit_id_pudl found in that plant. Mark that generator with a label
+    in the bga_source column consisting of label_prefix + the prime mover code.
+
+    If fuel_type_code_pudl is not None, then only assign new Unit IDs to those
+    generators having the specified fuel type code, and use that fuel type code
+    as the label prefix, e.g. "coal_st" for a coal-fired steam turbine.
+
+    Only generators having NA unit_id_pudl will be assigned a new ID.
+
+    Args:
+        gens_df (pandas.DataFrame): A collection of EIA generator records.
+            Must include the ``plant_id_eia``, ``generator_id`` and
+            ``prime_mover_code`` and ``unit_id_pudl`` columns.
+        prime_mover_codes (list): List of prime mover codes for which we are
+            attempting to assign simple Unit IDs.
+        fuel_type_code_pudl (str, None): If not None, then limit the records
+            assigned a unit_id to those that have the specified
+            fuel_type_code_pudl (e.g. "coal", "gas", "oil", "nuclear")
+        label_prefix (str): String to use in labeling records as to how their
+            unit_id_pudl was set. Will be concatenated with the prime mover
+            code.
+
+    Returns:
+        pandas.DataFrame: A new dataframe with the same rows and columns as
+        were passed in, but with the unit_id_pudl and bga_source columns updated
+        to reflect the newly assigned Unit IDs.
+
+    """
+    if fuel_type_code_pudl is not None:
+        # Need to make this only apply to consistent inter-year fuel types.
+        fuel_type_mask = gens_df.fuel_type_code_pudl == fuel_type_code_pudl
+        # If we're selecting based on fuel as well as prime mover, use that
+        # in the label
+        logger.info("Using fuel type %s as label_prefix.", fuel_type_code_pudl)
+        label_prefix = fuel_type_code_pudl
```

Comment:
By "this" do you mean the `label_prefix`? Yes. If you give it a `label_prefix` and a `fuel_type_code_pudl` you'll get the confusing behavior of losing your label prefix, which is why I logged it. Probably this isn't great. The labels as a whole are also quite ad hoc at this point. I could raise an exception instead of they're both set.

Reply:
It turned out the same behavior could be gotten by changing the arguments so I removed this to avoid confusing behavior.